src/p_range.jl

```diff
@@ -552,7 +552,7 @@ interface.
 -  `periodic::NTuple{N}=ntuple(i->false,N)`: Use or not periodic boundaries per direction.
 
 For convenience, one can also provide scalar inputs instead tuples
-to create 1D block partitions. In this case, the argument `np` can be omitted
+to create 1D block partitions. In this case, the argument `np` is omitted
```

Comment:
This looks like a breaking change. Is it really needed? Please keep the current option of not omitting np.

Reply:
I had to remove this option because this would otherwise result i a duplicate function signature.

```julia
uniform_partition(rank,n::Integer,ghost::Integer,periodic::Bool=false)
```
would be the same as
```julia
uniform_partition(rank,np::Integer,n::Integer)
```
if periodic is not specified.

I'm open to other ideas on how to handle this problem

Reply:
Hi @Joroks,

good point. I see the problem. 

Maybe we can add a new method in which the optional arguments are key-word arguments:

```julia
function uniform_partition(ranks,np::Tuple,n::Tuple;
    ghost=map(i->false,np),periodic=map(i->false,np))
end
```

This one should not interfere with any of the existing methods, I would say.

And for the moment build the thick ghost layer only using always tuples, also in the 1d case.

In the next breaking release, we can polish all these methods.

Reply:
The reason for this is that I don't want to introduce breaking changes at this moment.